mount.go

```diff
@@ -91,3 +95,43 @@ func getDeviceForPath(path string) (*device, error) {
 
 	return dev, nil
 }
+
+const (
+	procMountsFile = "/proc/mounts"
+
+	fieldsPerLine = 6
+)
+
+func getDevicePathAndFsType(mountPoint string) (string, string, error) {
+	if mountPoint == "" {
+		return "", "", fmt.Errorf("Mount Point cannot be empty")
+	}
+
+	file, err := os.Open(procMountsFile)
+	if err != nil {
+		return "", "", err
+	}
+
+	defer file.Close()
+
+	reader := bufio.NewReader(file)
+	for {
+		line, err := reader.ReadString('\n')
+		if err == io.EOF {
+			break
+		}
+
+		fields := strings.Fields(line)
+		if len(fields) != fieldsPerLine {
+			return "", "", fmt.Errorf("Incorrect no of fields (expected %d, got %d)) :%s", fieldsPerLine, len(fields), line)
+		}
+
+		// fields[0] => Device, fields[1] => Path , fields[2] => Type
```

Comment:
would having enums here help vs 0, 1, 2? You already have a constant definition above